compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -2556,6 +2556,11 @@ object Parsers {
       val clsName = modName.toTypeName
       val constr = classConstr()
       val templ = templateOpt(constr, isEnum = true)
```

Comment:
What about calling `template` here instead of `templateOpt`?

Reply:
`template` requires parents, but the template body can be empty.